grails-app/controllers/org/pih/warehouse/requisition/RequisitionTemplateController.groovy

```diff
@@ -86,6 +86,78 @@ class RequisitionTemplateController {
         }
     }
 
+    def exportPage = { }
+
+    def exportAsCsv = {
+        def requisitionItems = []
+        def origins = []
+        def destinations = []
+
+        params.origin.each { origin ->
+            origins << Location.get(origin)
+        }
+        params.destination.each { destination ->
+            destinations << Location.get(destination)
+        }
+
+        def requisitions = requisitionService.getRequisitionTemplates(origins, destinations)
+        requisitionService.getRequisitionTemplatesItems(requisitions).groupBy { it.product }.collect { product, value ->
+            requisitionItems << [
+                    product: product,
+                    productCode: product.productCode ?: "",
+                    productName: product.name ?: "",
+                    category: product?.category?.name ?: "",
+                    pricePerUnit: product?.pricePerUnit ?: "",
+            ]
+        }
+
+        def sw = new StringWriter()
+
+        try {
+            if (requisitionItems) {
+                sw.append(g.message(code: 'product.code.label')).append(",")
+                sw.append(g.message(code: 'product.description.label')).append(",")
+                sw.append(g.message(code: 'product.primaryCategory.label')).append(",")
+
+               requisitions.each { requisition ->
+                    String stocklistName = StringEscapeUtils.escapeCsv(requisition?.name)
+                    sw.append(stocklistName).append(",")
+                }
+
+                sw.append(g.message(code: 'product.unitCost.label')).append(",")
+                sw.append(g.message(code: 'product.totalValue.label'))
+
+                sw.append("\n")
+                requisitionItems.each { requisitionItem ->
+                    def totalCost = 0
+                    requisitions.each { requisition ->
+                        totalCost = requisitionItem.pricePerUnit ? totalCost + (requisition.getQuantityByProduct(requisitionItem.product) * requisitionItem.pricePerUnit) : ""
```

Comment:
I'll approve the PR since you did what I asked but this should probably be moved into the data transformation logic above so we keep it all in one place? Same with the quantity by requisition.

Reply:
@jmiranda changed it, thanks!